Game/MyProject/Game/SceneManagement/BoundingBox.cpp

```diff
@@ -0,0 +1,96 @@
+//
+//  BoundingBox.cpp
+//  RacingGame
+//
+//  Created by Tamas Both - (p) on 18/04/2019.
+//
+
+#include "BoundingBox.hpp"
+
+// --------------------------------
+// PUBLIC FUNCTION(S) AND METHOD(S)
+// --------------------------------
+
+// calculates the bounding box with given mesh and trasform matrix
+void BoundingBox::calculate(const MeshPtr& mesh, const IvMatrix44& transform) {
+    calculatePoints(mesh->getMinMaxVertices(), transform);
+}
+
+// calculates the bounding box with given bounding boxes
+void BoundingBox::calculate(const BoundingBox::Bounds& points) {
+    BorderPoint borders;
+    IvMatrix44  transform;
+    
+    // sets the starting values
+    borders.resize(3);
+    borders[0].x = points[0].x;
+    borders[0].y = points[0].x;
+    borders[1].x = points[0].y;
+    borders[1].y = points[0].y;
+    borders[2].x = points[0].z;
+    borders[2].y = points[0].z;
+    
+    // calculates the min and max for each axe
+    for (IvVector3 i : points) {
+        if (i.x < borders[0].x)
+            borders[0].x = i.x;
+        if (i.x > borders[0].y)
+            borders[0].y = i.x;
+        
+        if (i.y < borders[1].x)
+            borders[1].x = i.y;
+        if (i.y > borders[1].y)
+            borders[1].y = i.y;
+        
+        if (i.z < borders[2].x)
+            borders[2].x = i.z;
+        if (i.z > borders[2].y)
+            borders[2].y = i.z;
+    }
+    
+    calculatePoints(borders, transform);
+}
+
+const IvVector3& BoundingBox::getMax() const {
+    return _max;
+}
+
+const IvVector3& BoundingBox::getMin() const {
+    return _min;
+}
+
+const IvVector3& BoundingBox::getCenter() const {
+    return _center;
+}
+
+const BoundingBox::Bounds& BoundingBox::getPoints() const {
+    return _points;
+}
+
+// ---------------------------------
+// PRIVATE FUNCTION(S) AND METHOD(S)
+// --------------------------------
+
+void BoundingBox::calculatePoints(const BorderPoint& points, const IvMatrix44& transform) {
```

Comment:
This function needs to take only the min max vertices from the mesh not a std::vector.